### PR TITLE
Allow numeric values for GreaterThan and LessThan

### LINKED ISSUE #18: LessThan, LessThanOrEquals, GreaterThan, GreaterThanOrEquals can be something else than just an int type
Hi,

Parsley only allows the **type** to be **int** for the following constraints:
- LessThan
- LessThanOrEquals
- GreaterThan
- GreaterThanOrEquals

As can be seen from this code snippet of for example LessThan:
image

But according to the Symfony documentation, it can be more than an **int**:
image

https://symfony.com/doc/4.4/reference/constraints/LessThan.html#value
https://symfony.com/doc/4.4/reference/constraints/LessThanOrEqual.html#value
https://symfony.com/doc/4.4/reference/constraints/GreaterThan.html#value
https://symfony.com/doc/4.4/reference/constraints/GreaterThanOrEqual.html#value

Can you at least make the allowed type **numeric**, so floats can be used as well.

Thanks in Advance,

With kind regards,
Ruben

### PR DESCRIPTION
**Pull request by Matts**
Closes #18 

Feedback welcome!

Kindly add label `hacktoberfest-accepted` to this PR. More details on https://hacktoberfest.digitalocean.com/hacktoberfest-update

### CODE REVIEW

File: Constraint/Constraints/GreaterThan.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ protected function configureOptions(OptionsResolver $resolver): void
     {
         $resolver
             ->setRequired(['value'])
-            ->setAllowedTypes('value', ['int'])
+            ->setAllowedTypes('value', ['numeric', 'string'])
```
Comment (J-Ben87) [inline Constraint/Constraints/GreaterThan.php:48]: You shouldn't allow `string` type here: `numeric` type already handles values like `"5"` or `"5.2"`.

By allowing `string` type, values like `"foo"` would be accepted, which is not supported by Parsley.

File: Constraint/Constraints/LessThan.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ protected function configureOptions(OptionsResolver $resolver): void
     {
         $resolver
             ->setRequired(['value'])
-            ->setAllowedTypes('value', ['int'])
+            ->setAllowedTypes('value', ['numeric', 'string'])
```
Comment (J-Ben87) [inline Constraint/Constraints/LessThan.php:48]: Same as above.

File: Tests/Constraint/Constraints/GreaterThanOrEqualTest.php [hunk 3]
```diff
@@ -20,12 +20,13 @@ public function testInvalidConfiguration(): void
         $this->expectException(InvalidOptionsException::class);
 
         new ParsleyAssert\GreaterThanOrEqual([
-            'value' => '5',
+            'value' => [],
```
Comment (J-Ben87) [inline Tests/Constraint/Constraints/GreaterThanOrEqualTest.php:23]: You could assert that `'value' => 'foo',` throws the exception

File: Tests/Constraint/Constraints/GreaterThanTest.php [hunk 4]
```diff
@@ -20,12 +20,13 @@ public function testInvalidConfiguration(): void
         $this->expectException(InvalidOptionsException::class);
 
         new ParsleyAssert\GreaterThan([
-            'value' => '5',
+            'value' => [],
```
Comment (J-Ben87) [inline Tests/Constraint/Constraints/GreaterThanTest.php:23]: Same as above.

File: Tests/Constraint/Constraints/LessThanOrEqualTest.php [hunk 5]
```diff
@@ -20,12 +20,43 @@ public function testInvalidConfiguration(): void
         $this->expectException(InvalidOptionsException::class);
 
         new ParsleyAssert\LessThanOrEqual([
-            'value' => '10',
+            'value' => [],
```
Comment (J-Ben87) [inline Tests/Constraint/Constraints/LessThanOrEqualTest.php:23]: Same as above.

File: Tests/Constraint/Constraints/LessThanTest.php [hunk 6]
```diff
@@ -20,12 +20,13 @@ public function testInvalidConfiguration(): void
         $this->expectException(InvalidOptionsException::class);
 
         new ParsleyAssert\LessThan([
-            'value' => '10',
+            'value' => [],
```
Comment (J-Ben87) [inline Tests/Constraint/Constraints/LessThanTest.php:23]: Same as above.

**Review by J-Ben87 — CHANGES_REQUESTED** [review]
Hi Matts,

Thank you for submitting the PR. I only have a few improvements to suggest, otherwise that looks good to me :slightly_smiling_face:

**Comment by Matts** [comment]
> Hi Matts,
> 
> Thank you for submitting the PR. I only have a few improvements to suggest, otherwise that looks good to me 🙂

Fixed.

**Review by J-Ben87 — APPROVED** [review]

**Comment by J-Ben87** [comment]
Thank you for contributing :slightly_smiling_face: 
Your contribution will be available as version 2.1.2 in Packagist.

**Pull request merged**